pkg/validate/dynamic/platformworkloadidentityprofile.go

```diff
@@ -0,0 +1,108 @@
+package dynamic
+
+import (
+	"context"
+	"net/http"
+
+	sdkauthorization "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v3"
+	"github.com/Azure/go-autorest/autorest/azure"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/armauthorization"
+	"github.com/Azure/ARO-RP/pkg/util/rbac"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+func (dv *dynamic) ValidatePlatformWorkloadIdentityProfile(ctx context.Context, oc *api.OpenShiftCluster, platformWorkloadIdentityRoles []api.PlatformWorkloadIdentityRole, roleDefinitions armauthorization.RoleDefinitionsClient) error {
+	dv.log.Print("ValidatePlatformWorkloadIdentityProfile")
+
+	platformIdentitiesActionsMap := map[string][]string{}
+
+	requiredOperatorIdentities := []string{}
+	recievedOperatorIdentities := []string{}
+	for _, role := range platformWorkloadIdentityRoles {
+		requiredOperatorIdentities = append(requiredOperatorIdentities, role.OperatorName)
+	}
+	for _, role := range oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
+		recievedOperatorIdentities = append(recievedOperatorIdentities, role.OperatorName)
+	}
+	ok := stringutils.ElementsMatch(requiredOperatorIdentities, recievedOperatorIdentities)
+	if !ok {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodePlatformWorkloadIdentityMismatch,
```

Comment:
Can we also return the required identities here? Something like `"properties.ValidatePlatformWorkloadIdentityProfile.PlatformWorkloadIdentities", "There's a mismatch between the required and expected set of platform workload identities for the requested OpenShift version '%s'. The required identities are '%v'", oc.Properties.ClusterProfile.Version, requiredOperatorIdentities)`
Or something like that.